src/app/portlets/content-types/fields/content-type-fields-variables/content-type-fields-variables.component.ts

```diff
@@ -0,0 +1,138 @@
+            this.initTableData();
+        }
+    }
+
+    /**
+     * Handle Delete event
+     * @param {FieldVariable} variable
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    deleteVariable(variable: FieldVariable): void {
+        this.fieldVariablesService.deleteFieldVariables(this.field.typeId, this.field.id, variable.id).subscribe(() => {
+            this.fieldVariables = this.fieldVariables.filter((item: FieldVariable) => item !== variable);
+        });
+    }
+
+    /**
+     * Handle Save event
+     * @param {FieldVariable} variable
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    saveVariable(variable: FieldVariable): void {
+        this.fieldVariablesService.saveFieldVariables(this.field.typeId, this.field.id, variable)
+            .subscribe((savedVariable: FieldVariable) => {
+                variable = savedVariable;
+                this.initTableData();
+            });
+    }
+
+    /**
+     * Handle table input's placeholder when an input has been changed succesfully
+     * @param {FieldVariable} variable
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    changeVariable(variable: any): void  {
+        this.handlePlaceholderInput(
+            variable,
+            '',
+            '',
+            this.messages['contenttypes.field.variables.key_placeholder.label'],
+            this.messages['contenttypes.field.variables.value_placeholder.label']);
+    }
+
+    /**
+     * Handle table input's placeholder when an input get focus
+     * @param {FieldVariable} variable
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    changeInitVariable(variable: any): void  {
+        this.handlePlaceholderInput(
+            variable,
+            this.messages['contenttypes.field.variables.key_placeholder.label'],
+            this.messages['contenttypes.field.variables.value_placeholder.label'],
+            '',
+            '');
+    }
+
+    /**
+     * Handle table input's placeholder when an input lost focus
+     * @param {FieldVariable} variable
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    changeCancelVariable(variable: any): void  {
+        this.handlePlaceholderInput(
+            variable,
+            '',
+            '',
+            this.messages['contenttypes.field.variables.key_placeholder.label'],
+            this.messages['contenttypes.field.variables.value_placeholder.label']);
+    }
+
+    // tslint:disable-next-line:cyclomatic-complexity
+    private handlePlaceholderInput(variable: any, compareKey: string, compareValue: string, keyLbl: string, valLbl: string): void {
+        variable.data.key = variable.data.key === compareKey && variable.column.field === 'key' ? keyLbl : variable.data.key;
+        variable.data.value = variable.data.value === compareValue && variable.column.field === 'value' ? valLbl : variable.data.value;
+    }
+
+    private initTableData(): void {
```

Comment:
Everytime we add, edit or remove a field we have to make a request to get them all again?